CMakeLists.txt

```diff
@@ -51,11 +51,14 @@ list(APPEND LIBS ${DEATHKNELL})
 
 find_library(TCMALLOC tcmalloc PATHS /usr/local/probe/lib )
 # Appended first instead
+find_library(NANO_MSG nanomsg PATHS /usr/local/lib )
+list(APPEND LIBS ${NANO_MSG})
 
-
+find_library(MPI mpi PATHS /usr/lib64/openmpi/lib )
+find_library(MPI_CXX mpi_cxx PATHS /usr/lib64/openmpi/lib )
 
 MESSAGE(" PROJECT_SRC = : ${PROJECT_SRC}")
-include_directories(${PROJECT_SRC})
+include_directories(${PROJECT_SRC} /usr/include/openmpi-x86_64/)
```

Comment:
@KjellKod as i explained, this is a work in progress. I will remove the mpi part after I am finished, unless we want to keep the process benchmark.